addons/survey/models/survey_survey.py

```diff
@@ -60,6 +59,24 @@ def _get_default_access_token(self):
     answer_count = fields.Integer("Started", compute="_compute_survey_statistic")
     answer_done_count = fields.Integer("Completed", compute="_compute_survey_statistic")
 
+    # scoring and certification fields
+    scoring_type = fields.Selection([
+        ('no_scoring', 'No scoring'),
+        ('scoring_with_answers', 'Scoring with answers at the end'),
+        ('scoring_without_answers', 'Scoring without answers at the end')],
+        string="Scoring", required=True, default='no_scoring')
+    passing_score = fields.Float('Passing score (%)', required=True, default=80.0)
+    limit_number_of_attempts = fields.Boolean('Limited number of attempts',
```

Comment:
Same here: I would go for `is_attempt_limited` and `attempt_limit`.
If 'is_limited' is checked, then the limit should be positive. No default value, as this one should be set by an onchange.

Reply:
Done.